Source/ProgressionSystemRuntime/Public/Widgets/PSMenuWidget.h

```diff
@@ -39,6 +39,18 @@ class PROGRESSIONSYSTEMRUNTIME_API UPSMenuWidget : public UUserWidget
 	 * May be called multiple times due to adding and removing from the hierarchy. */
 	virtual void NativeConstruct() override;
 
+	/** Called when the current game state was changed. */
+	UFUNCTION(BlueprintNativeEvent, BlueprintCallable, Category = "C++", meta = (BlueprintProtected))
+	void OnGameStateChanged(ECurrentGameState CurrentGameState);
+
+	/** Subscribes to the end game state change notification on the player state. */
+	UFUNCTION(BlueprintNativeEvent, BlueprintCallable, Category = "C++", meta = (BlueprintProtected))
+	void OnLocalPlayerStateReady(AMyPlayerState* PlayerState, int32 CharacterID);
+
+	/** Called when the end game state was changed. */
```

Comment:
(Optional) You can ignore this comment. In future, please try always specify the reason why you use any function, when possible.

Before:
```
/** Called when the end game state was changed. */
```

After:
```
/** Called when the end game state was changed to toggle progression widget to player. */
```

Reply:
done https://github.com/h4rdmol/ProgressionSystem/pull/41/commits/4d55a07fe39d5cfa2f5d80f3e9f35929f46cc83d